buildinfo_test.go

```diff
@@ -658,6 +658,73 @@ func TestBuildAddGitEnvBuildNameAndNumber(t *testing.T) {
 	testBuildAddGit(t, true)
 }
 
+func TestBuildPublishWithOverwrite(t *testing.T) {
+	initArtifactoryTest(t, "")
+	buildName := "overwrite-test-build"
+	buildNumber := "1"
+	preReleaseBuildNumber := "1-rc"
+	defaultNumberOfBuilds := 5
+
+	// Clean old build tests if exists
+	inttestutils.DeleteBuild(serverDetails.ArtifactoryUrl, buildName, artHttpDetails)
+
+	// Publish build info without overwrite flag
+	for i := 0; i < defaultNumberOfBuilds; i++ {
+		runRt(t, "bp", buildName, buildNumber)
+	}
+	for i := 0; i < 2; i++ {
+		runRt(t, "bp", buildName, preReleaseBuildNumber)
+	}
+	publishedBuildInfo, found, err := tests.GetBuildRuns(serverDetails, buildName, buildNumber)
+	assert.NoError(t, err)
+	assert.True(t, found, "build info was expected to be found")
+	// Verify if total of 7 build info's are available
+	assert.Equal(t, 7, len(publishedBuildInfo.BuildsNumbers), "expected five build info's to be available")
+
+	buildNumberFrequency := calculateBuildNumberFrequency(publishedBuildInfo)
+	// Verify if 5 build info's are available for given build number and 2 for pre-release build number
+	assert.Equal(t, defaultNumberOfBuilds, buildNumberFrequency[buildNumber])
+	assert.Equal(t, 2, buildNumberFrequency[preReleaseBuildNumber])
+
+	// Publish build info with overwrite flag
+	runRt(t, "bp", buildName, buildNumber, "--overwrite=true")
+	publishedBuildInfo, found, err = tests.GetBuildRuns(serverDetails, buildName, buildNumber)
+	assert.NoError(t, err)
+	assert.True(t, found, "build info was expected to be found")
+	buildNumbersFrequencyAfterOverwrite := calculateBuildNumberFrequency(publishedBuildInfo)
+	// Since overwrite is ran for buildNumber verify if only one build info is available for given build number
+	assert.Equal(t, 1, buildNumbersFrequencyAfterOverwrite[buildNumber])
+	// Since overwrite is ran for buildNumber verify no change for preReleaseBuildNumber
+	assert.Equal(t, 2, buildNumbersFrequencyAfterOverwrite[preReleaseBuildNumber])
```

Comment:
This code block is kind of duplicated for every test case, maybe a common function would be useful

Reply:
I tried moving lot of them into a common function and when I read test case whats actually happening is abstracted into common function making it less readable how ever I moved the common part of assert.NoError and assert.True into a common function which is very much readable